static/app/components/devtoolbar/components/replay/replayPanel.tsx

```diff
@@ -0,0 +1,108 @@
+import {useContext, useState} from 'react';
+import {css} from '@emotion/react';
+
+import {Button} from 'sentry/components/button';
+import AnalyticsProvider, {
+  AnalyticsContext,
+} from 'sentry/components/devtoolbar/components/analyticsProvider';
+import SentryAppLink from 'sentry/components/devtoolbar/components/sentryAppLink';
+import useReplayRecorder from 'sentry/components/devtoolbar/hooks/useReplayRecorder';
+import {resetFlexRowCss} from 'sentry/components/devtoolbar/styles/reset';
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import {IconPause, IconPlay} from 'sentry/icons';
+import type {PlatformKey} from 'sentry/types/project';
+
+import useConfiguration from '../../hooks/useConfiguration';
+import {panelInsetContentCss, panelSectionCss} from '../../styles/panel';
+import {smallCss} from '../../styles/typography';
+import PanelLayout from '../panelLayout';
+
+const TRUNC_ID_LENGTH = 16;
+
+export default function ReplayPanel() {
+  const {trackAnalytics} = useConfiguration();
+
+  const {disabledReason, isDisabled, isRecording, lastReplayId, start, stop} =
+    useReplayRecorder();
+
+  const {eventName, eventKey} = useContext(AnalyticsContext);
+  const [buttonLoading, setButtonLoading] = useState(false);
+  return (
+    <PanelLayout title="Session Replay">
+      <Button
+        size="sm"
+        icon={isRecording ? <IconPause /> : <IconPlay />}
+        disabled={isDisabled || buttonLoading}
+        title={disabledReason}
+        onClick={async () => {
+          setButtonLoading(true);
+          const success = isRecording ? await stop() : await start();
+          setButtonLoading(false);
+          const type = isRecording ? 'stop' : 'start';
+          trackAnalytics?.({
+            eventKey: eventKey + `.${type}-button-click`,
+            eventName: eventName + `${type} button clicked`,
+          });
+          if (!success) {
+            trackAnalytics?.({
+              eventKey: eventKey + `${type}-button-failed`,
+              eventName: eventName + `${type} button failed`,
+            });
+          }
+        }}
+      >
+        {isRecording
+          ? 'In progress. Click to stop recording'
+          : 'Start recording the current session'}
```

Comment:
There's potentially a third state where it's buffering

Reply:
Should we surface this to users? Whether it's buffering or not